app/controllers/api/v1/orders_controller.rb

```diff
@@ -40,6 +41,19 @@ def restore
 
         render :json => order
       end
+
+      private
+
+      def service_offering_check
```

Comment:
@eclarizio I can see why this `before_action` filter method was placed here, but it does break our current practice of not adding methods using logic ( apart from what is needed to call a service from the action itself ) into our skinny controllers. 
It also is setting an instance variable which breaks the current practice of returning a boolean.

Possibly this logic should be moved into a mixin.

Reply:
@syncrou Yeah, this is true, I was trying to figure out the best way to do it without passing the ID of the order around so that we don't have to look it up twice. I'm not sure what you mean by current practice of returning a boolean, as far as I can see, the `RBACMixin` that utilizes most of our `before_action` filters either doesn't return anything or raises errors.

I can move it to a mixin, but I'd prefer to still set the instance variable so we don't have to hit the db twice. Sound ok?

Reply:
@eclarizio yes, that will work.